src/ReactiveSocket.cpp

```diff
@@ -25,19 +25,29 @@
 namespace reactivesocket {
 
 ReactiveSocket::~ReactiveSocket() {
-  stats_.socketClosed();
+  close();
+}
+
+void ReactiveSocket::close() {
   // Force connection closure, this will trigger terminal signals to be
   // delivered to all stream automata.
   connection_->disconnect();
 }
 
+void ReactiveSocket::onClose(std::unique_ptr<CloseCallback> closeCallback) {
+  // unimplemented
+  CHECK(false);
+  //  connection_->onClose(std::move(closeCallback));
+}
+
 std::unique_ptr<ReactiveSocket> ReactiveSocket::fromClientConnection(
     std::unique_ptr<DuplexConnection> connection,
     std::unique_ptr<RequestHandler> handler,
-    Stats& stats) {
+    Stats& stats,
+    uint32_t keepAliveDelay) {
   std::unique_ptr<ReactiveSocket> socket(new ReactiveSocket(
```

Comment:
can we replace calling "new ReactiveSocket" with std::make_unique ?
I like to keep following the guideline that if there is "new", there should be "delete". If I don't have new, then I don't have to search for "delete". Functionally, I know it means the same.